src/Fields/field_boundary_buffers.jl

```diff
@@ -132,6 +132,25 @@ function fill_send_buffers!(c::OffsetArray, buff::FieldBoundaryBuffers, grid, ::
     return nothing
 end
 
+#####
+##### Single sided fill_send_buffers!
+#####
+
+fill_send_buffers!(c::OffsetArray, buff::FieldBoundaryBuffers, grid, ::Val{:west}) = 
+    _fill_west_send_buffer!(parent(c), buff, buff.west, halo_size(grid)[[1, 2]]..., size(grid)[[1, 2]]...)
+fill_send_buffers!(c::OffsetArray, buff::FieldBoundaryBuffers, grid, ::Val{:east}) = 
+    _fill_east_send_buffer!(parent(c), buff, buff.east, halo_size(grid)[[1, 2]]..., size(grid)[[1, 2]]...)
+fill_send_buffers!(c::OffsetArray, buff::FieldBoundaryBuffers, grid, ::Val{:south}) = 
+    _fill_south_send_buffer!(parent(c), buff, buff.south, halo_size(grid)[[1, 2]]..., size(grid)[[1, 2]]...)
+fill_send_buffers!(c::OffsetArray, buff::FieldBoundaryBuffers, grid, ::Val{:north}) = 
+    _fill_north_send_buffer!(parent(c), buff, buff.north, halo_size(grid)[[1, 2]]..., size(grid)[[1, 2]]...)
+fill_send_buffers!(c::OffsetArray, buff::FieldBoundaryBuffers, grid, ::Val{:bottom}) = nothing
+fill_send_buffers!(c::OffsetArray, buff::FieldBoundaryBuffers, grid, ::Val{:top}) = nothing
```

Comment:
What advantage is using `Val{:top}`?

Usually one would just write something like `fill_top_buffers!`.

Reply:
that function is needed here 
https://github.com/CliMA/Oceananigans.jl/blob/af8450fe65f993f89411fdfa8ace4b1a4bf291b2/src/DistributedComputations/halo_communication.jl#L184
where we do not necessarily know which direction is being filled so we infer it from the `fill_halo!` function

Reply:
Ok got it. My main comment then is remember that by YASGuide the function arguments have the ordering:

1. Anything being mutated (in a mutating function)
2. Functions being dispatched on (ie best not to bury dispatchable arguments deep in the argument list)
3. Everything else in a nice or logical order

We break this only for the operator form functions which need to have first arguments `i, j, k, grid`.

Reply:
Changed it